polkadot/xcm/xcm-builder/src/filter_asset_location.rs

```diff
@@ -26,7 +26,7 @@ pub struct NativeAsset;
 impl ContainsPair<Asset, Location> for NativeAsset {
 	fn contains(asset: &Asset, origin: &Location) -> bool {
 		log::trace!(target: "xcm::contains", "NativeAsset asset: {:?}, origin: {:?}", asset, origin);
-		matches!(asset.id, AssetId(ref id) if id == origin)
+		matches!(asset.id, AssetId(ref id) if id.chain_location() == *origin)
 	}
 }
```

Comment:
What is a "native" asset? We don't have a clear definition or complete consensus on any definition.

This PR changes the inferred meaning of `NativeAsset` of `origin` chain from:
```diff
- The one single asset whose location matches the `origin` chain location.
+ Any asset originating from sub-locations of the `origin` chain location,
+ i.e. originates from "somewhere" on that chain.
```

Whether this change is "correct" is subjective AFAICT.

But since this `NativeAsset` struct is not used by any Relay chain or any System Parachain, and is mostly used by ecosystem parachains, I can defer that decision to ecosystem engineers. Ergo, I'm approving this.

Reply:
I agree that the correctness of calling it "Native" is subjective, but I am also open to implement this as a separate struct. Although we would have to come up with a new terminology (ReserveAsset maybe?).

In any case I think some of the ecosystem projects (e.g., Moonbeam & Acala) represent their native token as a sub-location of their chain (appending a `PalletIndex` or `GeneralKey`)

Reply:
In terms of terminology, I think it is more common to refer to it as a reserve asset because of the concept of the "Reserve Chain".

I think we all agree that a reserve/native asset is an asset that originates in a reserve chain.

A broader question is, for example, Polkadot and AssetHub-Polkadot are both reserve chains for DOT; I guess this means that DOT is a reserve/native asset of both chains as well. How does System Parachain or the Relay Chain refer to its own asset? I think if it does not have (theoretically speaking) the concept of "NativeAsset," then the name "ReserveAsset" is more natural because there is a reserve chain for that asset.